### ISSUE TITLE
Unable to reproduce results on different systems

### ISSUE THREAD

**Issue by surbhi1944** [hill-a/stable-baselines#628]
Earlier i raised the issue that running same code on same seed multiple times shows different results.
That issue was in stable-baselines2.8. Now with stable-baselines2.9 that issue is solved and i am able to get same results on multiple runs on same seed.

Now the another issue i am facing with stable-baselines2.9 is that: if i run same code on same seed on two different laptops then it is giving different results. How to resolve this issue. Though such issue does not come when i tried github/garage code for DDPG.

--------------------------------------------------------------------------------------------------------------
                               stable-baselines2.9
--------------------------------------------------------------------------------------------------------------
Code:
difsed=[30]
for difsed in range(1):
    seed=sed[difsed]
    import gym
    import random
    import numpy as np
    import os
    import matplotlib.pyplot as plt
    import tensorflow as tf
    from stable_baselines.ddpg.policies import LnMlpPolicy
    from stable_baselines.bench import Monitor
    from stable_baselines.results_plotter import load_results, ts2xy
    from stable_baselines import DDPG
    from stable_baselines.ddpg import OrnsteinUhlenbeckActionNoise,AdaptiveParamNoiseSpec
    from datetime import datetime, timedelta
    import time
    from mpi4py import MPI
    from stable_baselines import logger
    from stable_baselines.common import set_global_seeds
    set_global_seeds(seed)
    os.environ['PYTHONHASHSEED']=str(seed)
    best_mean_reward, n_steps = -np.inf, 0
    def callback(_locals, _globals):
        global log_dir
        #print("4 my callbck ",np.random.get_state()[1][0])
        global n_steps, best_mean_reward
        # Print stats every 1000 calls3
        if (n_steps + 1) % 1000 == 0:
        # Evaluate policy training performance
            x, y = ts2xy(load_results(log_dir), 'timesteps')
            if len(x) > 0:
                mean_reward = np.mean(y[-100:])
                #print(x[-1], 'timesteps')
                #print("Best mean reward: {:.2f} - Last mean reward per episode: {:.2f}".format(best_mean_reward, mean_reward))
                # New best model, you could save the agent here
                if mean_reward > best_mean_reward:
                    best_mean_reward = mean_reward
                    # Example for saving best model
                    #print("Saving new best model")
                    _locals['self'].save(log_dir + 'best_model.pkl')
        n_steps += 1
        # Returning False will stop training early
        return True

    path="/home/admin12/Documents/stable-baselines/results/ddpg"
    environment=["LunarLanderContinuous-v2"]
    ts=[int(1e6)]
    for name in range(len(environment)):
        exp="seeding"
        log_dir = path+"/"+environment[name]+"/"+exp+"/"+str(seed)+"/"
        os.makedirs(log_dir, exist_ok=True)
        env = gym.make(environment[name])
        env.seed(seed)
        env.action_space.seed(seed)
        env.observation_space.seed(seed)
        env =  Monitor(env, log_dir, allow_early_resets=True)
        n_actions = env.action_space.shape[-1]
        action_noise =OrnsteinUhlenbeckActionNoise(mean=np.zeros(n_actions), sigma=float(0.2) * np.ones(n_actions))
        #param_noise = None
        param_noise = AdaptiveParamNoiseSpec(initial_stddev=0.1, desired_action_stddev=0.1)
        model = DDPG(LnMlpPolicy, env, param_noise=param_noise,batch_size=64,buffer_size=1000000,enable_popart=False, action_noise=action_noise, verbose=4,seed=seed,n_cpu_tf_sess=1)
        start_time=time.time()
        model.learn(total_timesteps=ts[name], callback=callback,log_dir =log_dir)
        model.save(log_dir + "ddpg_"+environment[name])
        print("seed  ",np.random.get_state()[1][0])

      

System 1:
![image](https://user-images.githubusercontent.com/51375621/71008456-a6356f80-210e-11ea-858f-351a48a86d2d.png)

System 2:
![image](https://user-images.githubusercontent.com/51375621/71008551-cd8c3c80-210e-11ea-9b4e-cdde85f1b4db.png)

Please tell how to fix this.

Thankyou

**Comment by Miffyli** [comment]
On first glance I would first point the finger at different underlying libraries, starting from lower level math libraries up to Tensorflow/CUDA versions. Did I understand you correctly some other code (DDPG part) manages to get same results on different machines with same seed? Could you provide more information on this (what code exactly, how did you run the DDPG code) as well as your machine specifications and library versions (`pip freeze` would suffice for a start).

**Comment by surbhi1944** [comment]
Thanks
The problem is solved when i tried v10 commit.
